Add configuration to replace sections of path strings according to regexes, and tags spans accordingly (#20)
zopaUK/ktor-opentracing: README.md

## Configuration 

### filter
Your application might be serving static content (such as k8s probes), for which you do not want to create traces. 
You can filter these out as follows:
```kotlin
install(OpenTracingServer) {
    filter { call -> call.request.path().startsWith("/_probes") }
}
```

### replaceInPathAndTagSpan

> When an API path contains some kind of id, it can be helpful to replace it with a constant string, so that similar spans

Reviewer: ```suggestion
When a request path contains an id, you can replace it with a constant string in the span operation name. This ensures that requests for different ids have the same span operation name. 
```

> are named the same and grouped together. The contents of the id is then tagged so that information is not lost.

Reviewer: The value of the id is then tagged on the span.

```kotlin
install(OpenTracingServer) {

>     tagAndReplace("customId", Regex("""[0-9]{8}-[0-9]{4}"""))

Reviewer 2: Not sure the name describes exactly what the function is doing here. What would you think of `extractTagFromPath` or `groupPathBy`?

}
```

> In the above example, `/path/12345678-1234` would be recorded as `/path/<customId>` with the tag `customId=12345678-1234`.

Reviewer: ```suggestion
In the above example, `/path/12345678-1234` would lead to a span named as `/path/<customId>` with the tag `customId=12345678-1234`.
```

> Note that UUIDs are already tagged and replaced by default.

Reviewer: ```suggestion
Note that UUIDs are already tagged and replaced by default with `UUID`.
```

### Filter Requests

> Your application might be serving static content (such as k8s probes), for which you do not want to create traces. 

Reviewer: good catch
